pkg/store/disk.go

```diff
@@ -61,35 +60,25 @@ func InspectDisk(diskName string) (*Disk, error) {
 	return disk, nil
 }
 
-// inspectDisk attempts to inspect the disk size and format by itself,
-// and falls back to inspectDiskWithQemuImg on an error.
+// inspectDisk attempts to inspect the disk size and format with qcow2reader.
 func inspectDisk(fName string) (size int64, format string, _ error) {
 	f, err := os.Open(fName)
 	if err != nil {
-		return inspectDiskWithQemuImg(fName)
+		return -1, "", err
 	}
 	defer f.Close()
 	img, err := qcow2reader.Open(f)
 	if err != nil {
-		return inspectDiskWithQemuImg(fName)
+		return -1, "", err
 	}
 	sz := img.Size()
 	if sz < 0 {
-		return inspectDiskWithQemuImg(fName)
+		return -1, "", err
```

Comment:
This doesn't work because `err` is `nil` here, and the caller only checks for `err` and not for `size < 0` as an error condition. So it should be something like

```suggestion
		return -1, "", fmt.Errorf("cannot determine size of %q", fName)
```

Reply:
yes, of course, sorry!